workflows/differentialabundance.nf

```diff
@@ -454,6 +474,37 @@ workflow DIFFERENTIALABUNDANCE {
             .mix(GSEA_GSEA.out.versions)
     }
 
+    if (params.gprofiler2_run) {
+
+        // For gprofiler2, use only features that are considered differential
+        ch_filtered_diff = FILTER_DIFFTABLE.out.filtered
+
+        if (!params.gprofiler2_background_file) {
+            // If param not set, use empty list as "background"
+            ch_background = []
+        } else if (params.gprofiler2_background_file == "auto") {
```

Comment:
I've been thinking about this, and I think that unless you have good reason for doing this, the background file should be the universe of genes considered for differential expression, which is the output of `CUSTOM_MATRIXFILTER`.

Reply:
Aah sure, you mean invert the default so that the user has to expressly deactivate the background? I think that makes sense!

Reply:
Yes- I really thing the background is the right way to go, not doing that is a really common mistake and one we should very strongly discourage. 

The universe of expressed genes is the correct background to use (which we effectively derive in matrix filtering), so I think using CUSTOM_MATRIXFILTER makes sense.